mlir/lib/Analysis/SliceAnalysis.cpp

```diff
@@ -115,6 +116,19 @@ static void getBackwardSliceImpl(Operation *op,
     }
   }
 
+  // Visit values that are defined above.
+  if (!options.omitUsesFromAbove) {
+    visitUsedValuesDefinedAbove(op->getRegions(), [&](OpOperand *operand) {
+      if (Operation *definingOp = operand->get().getDefiningOp()) {
```

Comment:
Should we check `backwardSlice->count(definingOp)` before calling `getBackwardSliceImpl()`?